internal/pkg/scaffold/helm/go_mod.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helm
+
+import (
+	"fmt"
+
+	"github.com/operator-framework/operator-sdk/internal/pkg/scaffold/input"
+	"github.com/operator-framework/operator-sdk/internal/pkg/scaffold/internal/deps"
+)
+
+const GoModFile = "go.mod"
+
+// GoMod - the go.mod file for a Helm hybrid operator.
+type GoMod struct {
+	input.Input
+}
+
+func (s *GoMod) GetInput() (input.Input, error) {
+	if s.Path == "" {
+		s.Path = GoModFile
+	}
+	s.TemplateBody = goModTmpl
+	return s.Input, nil
+}
+
+const goModTmpl = `module {{ .Repo }}
+
```

Comment:
`go mod` supports enforcing the go version with the compiler, so if we want, we can add a line with `go 1.12` to our go mod files.

Reply:
Because modules are valid for go 1.11+ I'd rather let users define that for themselves if they want.

Reply:
That's fine. I just noticed other areas in the docs where we reference depending on go 1.12. Should we change the docs to say go 1.11+?

The reason I even noticed this is that `go mod init` [populates the required go version](https://github.com/golang/go/blob/master/src/cmd/go/internal/modload/init.go#L438-L452) in the `go.mod` file it creates based on the installed go version.

Reply:
Yeah I'll update the docs to specify go 1.11+.

Looks like go version is a new feature in go 1.12. I say we leave this alone for now. We can emulate `go mod init` behaviour when we support modules by default, i.e. when 1.13 is released.